File: TableViewAgent/AdditionalCellState/AdditionalCellStateHideEditing.m
//
//  AdditionalCellStateHideEditing.m
//  TableViewAgent
//
//  Created by P.I.akura on 2013/06/18.
//  Copyright (c) 2013年 P.I.akura. All rights reserved.
//

#import "AdditionalCellStateHideEditing.h"

@implementation AdditionalCellStateHideEditing

- (BOOL)isShowAddCell:(BOOL) editing {
    return editing == NO;
}

- (ChangeInState)changeInState:(BOOL) editing {
    return (editing) ? ChangeInStateHide : ChangeInStateShow;
}

@end

File: lib/AdditionalCellState/AdditionalCellStateHideEditing.swift
//
//  AdditionalCellStateHideEditing.swift
//  TableViewAgent
//
//  Created by akuraru on 2014/07/29.
//  Copyright (c) 2014年 P.I.akura. All rights reserved.
//

import Foundation

class AdditionalCellStateHideEditing : AdditionalCellState {
    override func isShowAddCell(editing :Bool) -> Bool {
        return !editing
    }
    
    override func changeInState(editing :Bool) -> Int {
        return (editing) ? 1 : 2;
    }
}